ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/Rejections.scala

```diff
@@ -24,22 +24,24 @@ private[transaction] class Rejections(metrics: Metrics) {
 
   private final val logger = ContextualizedLogger.get(getClass)
 
-  def buildRejectionStep[A](
+  def buildImmediateRejectionStep[A](
       transactionEntry: DamlTransactionEntrySummary,
       rejection: Rejection,
       recordTime: Option[Timestamp],
-  )(implicit loggingContext: LoggingContext): StepResult[A] = {
-    buildRejectionStep(
+  )(implicit loggingContext: LoggingContext): StepResult[A] =
+    buildImmediateRejectionStep(
       buildRejectionEntry(transactionEntry, rejection),
+      rejection.description,
       recordTime,
     )
-  }
 
-  def buildRejectionStep[A](
+  def buildImmediateRejectionStep[A](
       rejectionEntry: DamlTransactionRejectionEntry.Builder,
+      rejectionDescription: String,
       recordTime: Option[Timestamp],
-  ): StepResult[A] = {
+  )(implicit loggingContext: LoggingContext): StepResult[A] = {
     Metrics.rejections(rejectionEntry.getReasonCase.getNumber).inc()
+    logger.trace(s"Transaction rejected, $rejectionDescription.")
```

Comment:
Would pre-execution still call this step and have the log produced? Or does it follow another code path?

Reply:
No, pre-execution (i.e., when there is no record time, the code path is not super-obvious indeed) calls directly the function below as it pre-computes a rejection check for post-execution to potentially use (because the time check is done at post-execution).